src/main/java/chess/domain/board/Position.java

```diff
@@ -0,0 +1,147 @@
+package chess.domain.board;
+
+import chess.domain.piece.Team;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class Position {
+
+	public static final int MIN_POSITION = 1;
+	public static final int MAX_POSITION = 8;
+	
+	private static final String OVER_RANGE_ERROR = "체스판 범위를 벗어나는 입력입니다.";
+
+	private final int row;
+	private final int column;
+
+	private static final Map<Integer, Map<Integer, Position>> cachedPositions = new HashMap<>();
+
+	private Position(final int row, final int column) {
+		validateRange(row, column);
+		this.row = row;
+		this.column = column;
+	}
+
+	public static Position of(final int row, final int column) {
+		cachedPositions.putIfAbsent(row, new HashMap<>());
+		cachedPositions.get(row).putIfAbsent(column, new Position(row, column));
+		return cachedPositions.get(row).get(column);
+	}
+
+	private void validateRange(final int row, final int column) {
+		if (isOverRange(row, column)) {
+			throw new IllegalArgumentException(OVER_RANGE_ERROR);
+		}
+	}
+
+	private boolean isOverRange(final int row, final int column) {
+		return row < MIN_POSITION || row > MAX_POSITION || column < MIN_POSITION || column > MAX_POSITION;
+	}
+
+	public Position addDirection(Direction direction) {
+		int row = direction.addRow(this.row);
+		int column = direction.addColumn(this.column);
+		return Position.of(row, column);
+	}
+
+	public boolean isLinerMove(Position position) {
+		return this.row == position.row || this.column == position.column;
+	}
+
+	public boolean isDiagonalMove(Position position) {
+		return Math.abs(this.row - position.row) == Math.abs(this.column - position.column);
+	}
+
+	public List<Position> calculateMovableByDirection(List<Direction> directions) {
```

Comment:
```suggestion
	public List<Position> calculateMovableByDirections(List<Direction> directions) {
```

명확한 의미를 담아서 메서드 이름을 지어보죠!

Reply:
네! `getArrivalPositionsByDirections`으로 변경해 보았습니다

Reply:
오... 제가 제안드린 메서드 이름보다 훨씬 의미가 잘 드러나네요! 굳굳